src/main/java/io/xdb/core/process/ProcessOptions.java

```diff
@@ -1,14 +1,29 @@
 package io.xdb.core.process;
 
-import io.xdb.gen.models.ProcessIdReusePolicy;
+import static io.xdb.core.utils.ProcessUtil.DEFAULT_NAMESPACE;
+
+import com.google.common.base.Strings;
+import io.xdb.core.utils.ProcessUtil;
+import io.xdb.gen.models.ProcessStartConfig;
 import lombok.Builder;
-import lombok.Getter;
 
 @Builder
-@Getter
 public class ProcessOptions {
 
+    // If not set, use the default value.
+    private final String namespace;
     private final String type;
-    private final ProcessIdReusePolicy processIdReusePolicy;
-    private final Integer timeoutSeconds;
+    private final ProcessStartConfig processStartConfig;
+
+    public String getNamespace() {
+        return Strings.isNullOrEmpty(namespace) ? DEFAULT_NAMESPACE : namespace;
+    }
+
+    public String getType(final Class<? extends Process> processClass) {
```

Comment:
It maybe better to move it to Process class as a package private method, so that it doesn't requre to pass in any parameter. 

Passing a process class here is a bit strange

Reply:
Another idea: maybe it's even better to require passing in a Process instance when creating this ProcessOptions. 

In the Process interface:
```java
public interface Process{
 
   default getProcessOptions(){
         return ProcessOptions.builder(this).build.
   }  
}
```

Reply:
In lombok builder, the default value of each field is NULL. Adding `@NonNull` will make sure the specified field must be assigned with a non-null value at the RUNTIME.

Use `ProcessOptionsBuilder builder(final Class<? extends Process> processClass)` is a good idea to provide `ProcessOptions.builder(this.getClass()).build()`.

Lombok also provides another way to genetate the ProcessOptionsBuilder by `new ProcessOptions.ProcessOptionsBuilder()`, I didn't find a way to disable it. So when using the `new` method to generate the builder, users have to be aware the processClass must be set explicitly.

Reply:
> So when using the new method to generate the builder, users have to be aware the processClass must be set explicitly.
yeah that's true. That's why it would be nice if the field being required is clear. 
Though, usually when there is a helper build which is easier to use, users would not use the raw/harder one.